.github/workflows/build.yml

```diff
@@ -81,12 +113,22 @@ jobs:
     if: ${{ github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == 'ls1intum/Artemis' }}
     runs-on: ubuntu-latest
     steps:
+    # If this is a workflow_dispatch and a 'sha' is provided, override RAW_URL
+    - name: Override RAW_URL for workflow_dispatch
+      if: ${{ github.event_name == 'workflow_dispatch' && inputs.raw_url != '' }}
+      run: echo "RAW_URL=${{ inputs.raw_url }}" >> $GITHUB_ENV
     - name: Compute Tag
       uses: actions/github-script@v7
       id: compute-tag
       with:
         result-encoding: string
         script: |
+          if (context.eventName === "workflow_dispatch") {
+            // When the workflow_call is triggered by another workflow,
+            // the parent's payload is passed to the child workflow
+            // This is why we are using the inputs from the parent workflow
+            return context.payload.inputs.HELIOS_BUILD_TAG;
+          }
```

Comment:
I do not like the strong coupling to both Helios and the parent workflow here. This should just be passed via an input to the reusable workflow.